src/test/java/org/broadinstitute/hellbender/utils/pileup/ReadPileupUnitTest.java

```diff
@@ -490,4 +492,124 @@ public void testSimplePileupWithOffset(){
+            element2.getRead().setBaseQualities(new byte[] {j});
+            logger.debug("Test: fixing ({}) and ({})", element1, element2);
+            ReadPileup.fixPairOverlappingQualities(element1, element2);
+            Assert.assertEquals(element1.getQual(), QualityUtils.MAX_SAM_QUAL_SCORE);
+            Assert.assertEquals(element2.getQual(), 0);
+            element1.getRead().setBaseQualities(iArray);
+            logger.debug("Test: fixing ({}) and ({})", element3, element1);
+            ReadPileup.fixPairOverlappingQualities(element3, element1);
+            Assert.assertEquals(element3.getQual(), QualityUtils.MAX_SAM_QUAL_SCORE);
+            Assert.assertEquals(element1.getQual(), 0);
+        }
+        // Finally, check what happens if both are the maximum value
+        element1.getRead().setBaseQualities(new byte[] {Byte.MAX_VALUE});
+        logger.debug("Test: fixing ({}) and ({})", element3, element1);
+        ReadPileup.fixPairOverlappingQualities(element3, element1);
+        Assert.assertEquals(element3.getQual(), QualityUtils.MAX_SAM_QUAL_SCORE);
+        Assert.assertEquals(element1.getQual(), 0);
+    }
+
+    @Test
+    public void testFixOverlappingQualitiesDeletedElements() throws Exception {
+        // Create two reads with deletion
+        final GATKRead read = ArtificialReadUtils.createArtificialRead("1M1D1M");
+        read.setBases(new byte[] {'A', 'A'});
+        read.setBaseQualities(new byte[] {60, 60});
+        final GATKRead read2 = ArtificialReadUtils.createArtificialRead("1M1D1M");
+        read.setBases(new byte[] {'A', 'A'});
+        read2.setBaseQualities(new byte[] {40, 40});
+        // Creates PileupElement with a copy of the reads to keep the original read unmodified, because the read from
+        // the PileupElement is a reference to the Object passed to the constructor. This is required
+        // because fixing the overlapping qualities change the read stored into PileupElement and we have to check that
+        // it does not change anything from the read.
+        final GATKRead copy = read.deepCopy();
+        final GATKRead copy2 = read2.deepCopy();
+        final PileupElement deleted = new PileupElement(copy, 1, copy.getCigarElement(1), 1, 0);
+        final PileupElement normal = new PileupElement(copy2, 0, copy.getCigarElement(0), 0, 0);
+        ReadPileup.fixPairOverlappingQualities(deleted, normal);
+        // Copy reads are a reference to the read into the PileupElement, and thus should not be modified.
+        // Thus, we use them to test if it is not modified. This could be checked uncommenting the next two lines:
+        // Assert.assertSame(deleted.getRead(), copy);
+        // Assert.assertSame(normal.getRead(), copy2);
+        Assert.assertEquals(copy, read);
+        Assert.assertEquals(copy2, read2);
+        ReadPileup.fixPairOverlappingQualities(normal, deleted);
+        Assert.assertEquals(copy, read);
+        Assert.assertEquals(copy2, read2);
+    }
+
```

Comment:
I use exactly the same test `DataProvider` to create a `ReadPileup` with the two `PileupElement` and call `ReadPileup.fixOverlaps()`.  Is that enought?